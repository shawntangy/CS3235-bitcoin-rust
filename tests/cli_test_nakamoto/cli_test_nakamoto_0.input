{"Initialize":["{\n    \"all_blocks\": {\n        \"0\": {\n            \"header\": {\n                \"parent\": \"0\",\n                \"merkle_root\": \"0\",\n                \"timestamp\": 0,\n                \"block_id\": \"0\",\n                \"nonce\": \"0\",\n                \"reward_receiver\": \"GENESIS\"\n            },\n            \"transactions_block\": {\n                \"merkle_tree\": {\n                    \"hashes\": []\n                },\n                \"transactions\": [\n                    {\n                        \"sender\": \"GENESIS\",\n                        \"receiver\": \"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\n                        \"message\": \"SEND $299792458\",\n                        \"sig\": \"GENESIS\"\n                    }\n                ]\n            }\n        }\n    },\n    \"children_map\": {},\n    \"block_depth\": {\n        \"0\": 0\n    },\n    \"root_id\": \"0\",\n    \"working_block_id\": \"0\",\n    \"orphans\": {},\n    \"finalized_block_id\": \"0\",\n    \"finalized_balance_map\": {\n        \"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\": 299792458\n    },\n    \"finalized_tx_ids\": []\n}","{\n    \"pool_tx_ids\": [],\n    \"pool_tx_map\": {},\n    \"pool_tx_map_status\": {},\n    \"removed_tx_ids\": [],\n    \"last_finalized_block_id\": \"0\"\n}","{\n    \"neighbors\": [],\n    \"addr\": {\n      \"ip\": \"127.0.0.1\",\n      \"port\": 9011\n    },\n    \"miner_thread_count\": 4,\n    \"nonce_len\": 10,\n    \"difficulty_leading_zero_len\": 5,\n    \"difficulty_leading_zero_len_acc\": 5,\n    \"miner_thread_0_seed\": 142,\n    \"mining_reward_receiver\": \"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\n    \"max_tx_in_one_block\": 4\n}"]}
"RequestChainStatus"
"RequestNetStatus"
"RequestMinerStatus"
"RequestTxPoolStatus"
{"GetAddressBalance":"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ=="}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $100   // By Alice   // 1678775190859\"]","LfIm8ADzaeu4OhiRcTTGkWoOrVAfYGE+ZCoMJQfhxpMgzHh898jL/1yPRyVd09eB"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $10   // By Alice   // 1678775193218\"]","kxpte1MoekIywWffEKc2CHkq+q5JbDUXmqKO2+gdG038jCS+4MU1e0wKsy/IuAOa"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $1   // By Alice   // 1678775194435\"]","L3FCLAeL9KW7jUq6VxmHXzUOSyJzzyYtJlii3IrBCMH0Zt+LncaDcmcR5BWz2NOs"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $2   // By Alice   // 1678775195836\"]","pBbKj6q4u7rUp4fhtv5/CkOci76Muqh3N6ROXiqOAq34gxUMisQ9riWuDPQ8gNI1"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $20   // By Alice   // 1678775196659\"]","YlZQz70MHgDbYStnt48PBLdnEb+SG7G+oXUQFQ1lkC9JPXpItdqRjkpELQ0wiBzF"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $200   // By Alice   // 1678775197076\"]","kA3p4ZkspmGQa3zno2TlHCVzXoluUXR3KvetkJhNyZQ8nY44ojZo+OJgDxMP78BO"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3   // By Alice   // 1678775200728\"]","GusgsCi2kn8PKmOSwZEH8gABDTNVI+b1ZVRuQjwLRJwA1Xr8sllo30LOhOuaW66E"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $30   // By Alice   // 1678775203331\"]","WNhRQSRBGb4nUmAT5OVAKuseYA3Ji52b1Sj36g2p7dNDZDfBFtmzYwzNUQwOk5hr"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $300   // By Alice   // 1678775204148\"]","QH9nk17Iz61n3O1I98JRXcbTzmWrRd8Euq+vZkK+cofEg/1WwAYuW1Aje/E60XCb"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
{"PublishTx":["[\"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==\",\"MDgCMQDZDExOs97sRTnQLYtgFjDKpDzmO7Uo5HPP62u6MDimXBpZtGxtwa8dhJe5NBIsJjUCAwEAAQ==\",\"SEND $3000   // By Alice   // 1678775205689\"]","mUA2ZeKuC3aKrhjnTwEvZAIcGUytCV1EBBaR7E1Cv7qvyj7MZkUo0Qhts4xtbHUX"]}
"RequestChainStatus"
"RequestNetStatus"
"RequestMinerStatus"
"RequestTxPoolStatus"
{"GetAddressBalance":"MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ=="}
"Quit"
