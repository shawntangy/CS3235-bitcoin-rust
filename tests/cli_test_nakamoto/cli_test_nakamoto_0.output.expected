{"Notify":"# Waiting for IPC Requests ..."}
{"Notify":"[Main] Start receiving trans thread"}
"Initialized"
{"ChainStatus":{"#blocks":"1","#orphans":"0","finalized_id":"0","root_id":"0","working_depth":"0","working_id":"0"}}
{"NetStatus":{"#address":"ip: 127.0.0.1 port: 9011","#recv_msg":"0","#send_msg":"0"}}
{"MinerStatus":{"#thread":"0","difficulty":"0","is_running":"false"}}
{"TxPoolStatus":{"#pool_tx_map":"0"}}
{"AddressBalance":["MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==",299792458]}
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
"PublishTxDone"
{"ChainStatus":{"#blocks":"1","#orphans":"0","finalized_id":"0","root_id":"0","working_depth":"0","working_id":"0"}}
{"NetStatus":{"#address":"ip: 127.0.0.1 port: 9011","#recv_msg":"0","#send_msg":"0"}}
{"MinerStatus":{"#thread":"0","difficulty":"0","is_running":"false"}}
{"TxPoolStatus":{"#pool_tx_map":"10"}}
{"AddressBalance":["MDgCMQCqrJ1yIJ7cDQIdTuS+4CkKn/tQPN7bZFbbGCBhvjQxs71f6Vu+sD9eh8JGpfiZSckCAwEAAQ==",299792458]}
"Quitting"

